TROUBLESHOOTING.md

```diff
@@ -0,0 +1,163 @@
+# Troubleshooting
+
+## How the Ruby LSP activation works
+
+The Ruby LSP extension runs inside VS Code's NodeJS runtime, like any other VS Code extension. This means that
+environment variables that are properly set in your shell may not exist inside the NodeJS process. To run the LSP server
+with the same Ruby version as your projects, we need to properly set these environment variables, which is done by
+invoking your Ruby version manager.
+
+The extension runs a command using your shell's interactive mode, so that version managers configured in files such as
+`~/.zshrc` are picked up automatically. The command then exports the environment information into a JSON, so that we can
+inject it into the NodeJS process and appropriately set the Ruby version and gem installation paths.
+
+As an example, the activation script for `zsh` using `rbenv` as a version manager will look something like this:
+
+```shell
+# Invoke zsh using interactive mode (loads ~/.zshrc) to run a single command
+# The command is `rbenv exec ruby`, which automatically sets all relevant environment variables and selects the
+# specified Ruby version
+# We then print the activated environment as a JSON. We read that JSON from the NodeJS process to insert the needed
+# environment variables in order to run Ruby correctly
+/bin/zsh -ic 'rbenv exec ruby -rjson -e "puts JSON.dump(ENV)"'
+```
+
+After activating the Ruby version, we then proceed to boot the server gem (`ruby-lsp`). To avoid having users include
+the `ruby-lsp` in their `Gemfile`, we currently create a custom bundle under the `.ruby-lsp` directory inside your
+project. That directory contains another `Gemfile`, that includes the `ruby-lsp` gem in addition to your project's
+dependencies. This approach allows us to automatically detect which formatter your project uses and which gems we need
+to index for features such as go to definition.
+
+**Note**: we are working with the rubygems/bundler team to have this type of mechanism properly supported from within
+Bundler itself, which is currently being experimented with in a plugin called `bundler-compose`. Once `bundler-compose`
+is production ready, the entire custom bundle created under the `.ruby-lsp` directory will go away and we'll rely on
+Bundler to compose the LOAD_PATH including the `ruby-lsp` gem.
```

Comment:
If you want to use GH's special syntax for note, you need to put this section under quote too:
```suggestion
> **Note**: we are working with the rubygems/bundler team to have this type of mechanism properly supported from within
> Bundler itself, which is currently being experimented with in a plugin called `bundler-compose`. Once `bundler-compose`
> is production ready, the entire custom bundle created under the `.ruby-lsp` directory will go away and we'll rely on
> Bundler to compose the LOAD_PATH including the `ruby-lsp` gem.
```

If that's not the case, just ignore this comment 😄